### ISSUE TITLE
Fatal Error after coloring pulls

### ISSUE THREAD

**Issue by NobbNobb** [nnogga/MethodDungeonTools#85]
I tried to color my pulls. The first pull was no problem. The second pull worked aswell. When i colored the third pull all mobs from the first pull turned black and the second pull lost his color completly. After that things got silly. I tried to color the first pull again but the thrid pull got colored. So i decided to reload my UI. Reopening MDT caused following error

![mdterror](https://user-images.githubusercontent.com/6708925/54833585-a9561900-4cbe-11e9-8b28-3df68a7ff32b.PNG)

And the Addon looked like this:
![mdt](https://user-images.githubusercontent.com/6708925/54833908-4a44d400-4cbf-11e9-8ba3-c79da43c8bdf.PNG)

I'm not sure if i did something not normal. Like clicking out of the coloring tool or anything like that. Gonna reinstall the addon and try to reproduce the error to give more informations on that bug

**Comment by bht-ts** [comment]
If I remember right, this was caused by a broken configuration file. In my case I had the pulls 1 & 3 in my configuration, but the index of 3, was assumed to be 2, because it only had 2 pulls. And you can't index pull 2, because it is pull 3. 

1) Is this still a problem, or is MDT now running correctly?

2) If yes, I could once fix your configuration file, if you can tell me Dungeon & Preset name.

3) @nnogga perhaps we should build a "repair" function, that checks explicitly for this error and later db errors, that are known. This was caused by the EnsureDBTables(), but should have been fixed with 2.5.0. So it can't be the case for new presets, but could be the case for old presets.

**Comment by NobbNobb** [comment]
I managed to fix the fatal error. I just had to switch to a different dungeon which didn't had a broken preset, reload the UI and open MDT again. After that I could switch to the broken preset and just delete it. I did a test with a new preset but couldn't reproduce the problem which caused the fatal error.

Now I'm just left with not being able to color my pulls. Coloring one pull is fine. The moment I color a second pull the first colored pull gets back to default.
